torchx/c_src/torchx.cpp

```diff
@@ -58,7 +58,7 @@ inline std::string type2string(const torch::ScalarType type)
   }
 
 #define CATCH()                                              \
-  catch (c10::Error error)                                   \
+  catch (c10::Error &error)                                   \
```

Comment:
This one silences the vast majority of warnings, which all stemmed from a "catch by value" warning here.